### ISSUE TITLE
Add suport for Baum TASO module

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#7985]
There are lots of users worldwide who are working with TASO module, especially in professional environments. Here is a demonstration of this module:
https://www.youtube.com/watch?v=2T5d3dvEwpI

### Expected behavior:
NVDA should support this device because Cobra is not being developed anymore. At least Baum does not provide support any further. We could send you a test device if needed.

### Actual behavior:
NVDA does not have an implemented driver for this module.

### System configuration:
NVDA version:
All

NVDA Installed or portable:
Both
Other information:

Windows version:
All

Name and version of other software in use when reproducing the issue:
n/a

### Other questions:

Does the issue still occur after restarting your PC?
n/a

Have you tried any other versions of NVDA?
n/a

**Comment by leonardder** [comment]
Tightly related to #5162.

@Adriani90: Does the driver I mentioned in #5162 have support for this device? I've seen the telephone module working with it.

Note that there is one major issue with supporting these modules. They have a dedicated braille bar of a few characters. We've not yet found a satisfactory solution for this. I guess they should be left out of the 80 cells a VarioPro offers, and than could be filled programmatically.

I have suggested some changes to the people at Baum, I'm not sure whether they have implemented them all yet, and if not, it is likely that they will never do.

**Comment by Adriani90** [comment]
@Leonardder I will test it. It seems like it supports it. Could you please post your suggestions as isse / pr on their repo? I know some developers and Maybe we can include their input as well. Or they could contribute since they have worked at Baum. AS far as I can see, there are some unassigned gestures for NVDA. And there are some conflicts regarding scrolling the different modules separately. Is there any possibility to include the separate scrolling feature in NVDA? It might work if NVDA could recognize more than one braille display automatically at the same time.

 

 

 

Thanks alot for your good job.

 

 

Von: Leonard de Ruijter [mailto:notifications@github.com] 
Gesendet: Montag, 12. Februar 2018 07:07
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Mention <mention@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Add suport for Baum TASO module (#7985)

 

Tightly related to #5162 <https://github.com/nvaccess/nvda/issues/5162> .

@Adriani90 <https://github.com/adriani90> : Does the driver I mentioned in #5162 <https://github.com/nvaccess/nvda/issues/5162>  have support for this device? I've seen the telephone module working with it.

Note that there is one major issue with supporting these modules. They have a dedicated braille bar of a few characters. We've not yet found a satisfactory solution for this. I guess they should be left out of the 80 cells a VarioPro offers, and than could be filled programmatically.

I have suggested some changes to the people at Baum, I'm not sure whether they have implemented them all yet, and if not, it is likely that they will never do.

**Comment by Adriani90** [comment]
Hello,

as stated in #5162, the TASO works in NVDA with the shared driver. However, as we are in process to reach the main developer of this driver, it has to be considered that driver updates are possible from time to time. I hope the driver can be implemented in NVDA so that we can perform broad testing and provide feedback to Florin and NVDA developers.

**Comment by Adriani90** [comment]
this issue can be closed in case the driver will not be implemented in NVDA.

**Comment by leonardder** [comment]
I belief the Baum TASO module is a quite important part of the VarioPro eco system. Therefore I think it is ok to also keep track of module related development in #5162. Feel free to share any thoughts, questions or findings regarding the several modules in that issue. I'm closing this one.
